### PR TITLE
terraform - data provider integration

### PR DESCRIPTION
**Pull request by marcoceppi**
This PR is a WIP and needs testing coverage. This produces the scaffolding in order to have terraform integrated as a data provider / cloud in C7N and enable default filters like Value to be processed for `tf` resource'd policies.

### CODE REVIEW

File: tools/c7n_terraform/c7n_terraform/provider.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        """The object representation.
+
+        Returns:
+            str: The object representation.
+        """
+        return "<tf-session>"
+
+    def client(self, service_name, version, component, **kw):
+
+        return MagicMock()
```
Comment (kapilt) [inline tools/c7n_terraform/c7n_terraform/provider.py:28]: this seems odd, the client invocation style here is specific to gcp provider

**Comment by kapilt** [comment]
also needs to be added to makefile to ensure tests are being run

File: tools/c7n_terraform/c7n_terraform/provider.py (hunk 1 above)
Comment (marcoceppi) [reply tools/c7n_terraform/c7n_terraform/provider.py:28]: A lot of this branch was heavily influenced by the c7n_gcp module, which I used as my starting point. Knowing more now than I did a month ago when I started I'll do a second pass to clean up a lot of these things

Comment (marcoceppi) [reply tools/c7n_terraform/c7n_terraform/provider.py:28]: My biggest challenge is a lot of custodian assumes there's some form of client / session which implications of what methods are available in that session. Peeling that back has taken more effort than I expected

Comment (kapilt) [reply tools/c7n_terraform/c7n_terraform/provider.py]: there shouldn't be any client construction outside of the code in this provider, the rest of the cross provider machinery should just treat the session as an opaque blob.

**Pull request closed without merging**